src/Resources/views/CRUD/base_list.html.twig

```diff
@@ -283,7 +283,13 @@ file that was distributed with this source code.
         <div class="col-xs-12 col-md-12 sonata-filters-box" style="display: {{ admin.datagrid.hasDisplayableFilters ? 'block' : 'none' }}" id="filter-container-{{ admin.uniqid() }}">
             <div class="box box-primary" >
                 <div class="box-body">
-                    <form class="sonata-filter-form form-horizontal {{ admin.isChild and 1 == admin.datagrid.filters|length ? 'hide' : '' }}" action="{{ admin.generateUrl('list') }}" method="GET" role="form">
+                    <form
+                            class="sonata-filter-form form-horizontal {{ admin.isChild and 1 == admin.datagrid.filters|length ? 'hide' : '' }}"
+                            action="{{ admin.generateUrl('list') }}"
+                            method="GET"
+                            role="form"
+                            data-default-values="{{ admin.getDefaultFilterParameters()|json_encode }}"
```

Comment:
admin is an AdminInterface, so the method should be added to the interface.

But it should be done with BC, so maybe adding the data-default-values only if the method exists.
And in the jquery, keeping the old behavior if there is no data-default-values.

Reply:
Agreed

Reply:
Check is definied?

Reply:
`getDefaultFilterParameters` method may not exist in the admin variable since it's an AdminInterface and not always an AbstractAdmin.

In php we would have done 
```
if (method_exists($admin, 'getDefaultFilterParameters')) {
 ...
}
```
But in twig I'm not sure about the best way to do this.

Reply:
https://caniuse.com/?search=includes

Reply:
> https://caniuse.com/?search=includes

I think you wanted to answer to another message.

AFAIK, Sonata was working on Internet explorer. 

You could use 
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/indexOf
https://caniuse.com/?search=indexOf

includes is basically indexOf !== -1.

Reply:
yes. you are right. updated

Reply:
Can check via attribute
https://twig.symfony.com/doc/3.x/functions/attribute.html

Reply:
updated